### PR TITLE
Add input validation and tests for sum_of_digits

### PR DESCRIPTION
**Pull request by yuri-spizhovyi-mit**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/tests/test_sum_of_digits.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Test module for sum_of_digits function.
+
```
Comment (olumide-AI) [inline solutions/tests/test_sum_of_digits.py:5]: You can add a test categories if you like. Here is an example 
"""
Test Categories:
 Standard Cases:
   - Tests the function with typical, valid input values.
   - Includes single-digit and multi-digit numbers.

Edge Cases:
   - Tests boundary conditions and special cases.
   - Includes zero, the largest single-digit number, and very large numbers.

 Defensive Tests:
   - Tests the function's response to invalid or unexpected inputs.
   - Includes negative numbers and non-integer inputs.
   """

File: solutions/tests/test_sum_of_digits.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+from solutions.sum_of_digits import sum_of_digits
+
+
+class TestSumOfDigits(unittest.TestCase):
+    """Test suite for the sum_of_digits function."""
+
+    def test_single_digit(self):
+        """Test with a single-digit number."""
+        result = sum_of_digits(5)
```
Comment (olumide-AI) [inline solutions/tests/test_sum_of_digits.py:20]: I'm not sure if the result = sum_of digits(5) is needed instead you can just do it in one line like this self.assertEqual(sum_of_digits(5), 5)
i think this result = is creating redundancy in your test file just pas sum of digits in your assertion should be fine. I will ask Evan to confirm my opinion.

File: solutions/sum_of_digits.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+A module for recursively calculating the sum of digits of an integer.
+
+Module contents:
+    - sum_of_digits: A recursive function to calculate the sum of digits.
+
+Created on 2024-12-31
+Author: Yurii Spizhovyi
+"""
+
+
```
Comment (olumide-AI) [inline solutions/sum_of_digits.py:13]: I think "n" is not a descriptive enough variable name i suggest using "number", "value", "input_num" etc would be better in my opinion and according to the code review

File: solutions/sum_of_digits.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+A module for recursively calculating the sum of digits of an integer.
+
+Module contents:
+    - sum_of_digits: A recursive function to calculate the sum of digits.
+
```
Comment (olumide-AI) [inline solutions/sum_of_digits.py:8]: And also is your name yurii or yuriy i don't know if that was a typo

**Review by olumide-AI — COMMENTED** [review]
I have check marked the part i suggest changes. There's only minimal changes i suggest you looking at and i have left some single line comment regarding this. The code is well written and thought out. Good job Yuriy. You're killing it

File: solutions/sum_of_digits.py (hunk 4 above)
Comment (yuri-spizhovyi-mit) [reply solutions/sum_of_digits.py:8]: Yuriy is a translation that was done years ago in Google translate and it was automatically pulled from the gmail account. But, in my official documents the Yurii variant is used. So I think it would be better to use it for this project.

File: solutions/tests/test_sum_of_digits.py (hunk 2 above)
Comment (yuri-spizhovyi-mit) [reply solutions/tests/test_sum_of_digits.py:20]: good catch. Changed

File: solutions/tests/test_sum_of_digits.py (hunk 1 above)
Comment (yuri-spizhovyi-mit) [reply solutions/tests/test_sum_of_digits.py:5]: perfect. Added.

File: solutions/sum_of_digits.py (hunk 3 above)
Comment (yuri-spizhovyi-mit) [reply solutions/sum_of_digits.py:13]: agree. Changed to 'number'

**Review by Mithchell509 — APPROVED** [review]

**Review by olumide-AI — APPROVED** [review]
Changes look good Yuriy, i think you're good to merge
